torch/optim/functional.py

```diff
@@ -6,6 +6,13 @@
 
 # TODO: use foreach API in optim.functional to do all the computation
 
+def _make_sparse(grad, grad_indices, values):
```

Comment:
@vincentqb Could you review these changes?

Reply:
LGTM. What is the advantage of writing this in this way?

Reply:
It's mainly because nested functions are not supported by TorchScript, so I moved it out as a separate function. Also I feel like define a nested function inside a for loop is not a good practice